enhancements/cluster-logging/api-audit-log-policy.md

```diff
@@ -0,0 +1,234 @@
+- mount or copy the policy to the local file-system.
+- configure vector with an Exec source that runs the filter program as a separate process using the policy file.
+- vector reads stdout of the filter from the Exec source.
+- from there we route the events as normal
+
+Note: all of the above should support multiple distinct policies, so unique names for policy files etc.
+
+### Risks and Mitigations
+
+#### Data security
+
+No new sensitive data is exposed.
+The `ClusterLogForwarder` can already forward API server audit logs on OSD, ROSA classic and on-premise clusters.
+
+#### Run-time resource costs
+
+OSD and ROSA classic have been doing this type of filtering on managed clusters for years, no new problems are expected.
+
+#### Compliance and legal risks
+
+Incorrectly filtering out important audit events may be a compliance/legal problem.
+The customer creates their own filtering configuration, it needs to be clear that it is their responsibility to do it correctly.
+
+#### Roll-out
+
+Existing customers using the CLO add-on will be unaffected.
+The customer `cl/instance` and `clf/instance` resources will not be modified.
+The new features will be activated only when the customer modifies their configuration as described above.
+
+The existing SRE Splunk exporter SHOULD be replaced by separate CL/CLF resources in a separate namespace as soon as possible.
+Depends on [LOG-1343 Multiple forwarders](https://issues.redhat.com/browse/LOG-1343).
+
+However, the Splunk exporter MAY continue to run alongside customer CLO deployments using these new features, if necessary.
+Both can scrape from the same audit log files simultaneously, and perform separate filtering.
+
+**NOTE**: this is not true for Hypershift, [see hypershift proposal](hypershift-audit-logs.md)
+
+### Drawbacks
+
+- Increased scope for user error.
+- Additional CPU cost, latency (to be measured)
+- New dependency to be maintained for the log collector.
+
+## Design Details
+
+### Open Questions
+
+#### Form of audit policy
+
+The [Kubernetes Audit Policy File](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/) is expressed as a k8s "object" but not as a typical k8s "resource (no spec/status).
+
+Should we
+- Use it as a resource unmodified
+- Wrap it in a more typical spec/status resource
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+Nothing new.
+
+### Version Skew Strategy
+
+The same CLO operator will manage the customer and SRE log forwarding.
+The configuration is entirely separate, but if customer or SRE require new features, the same operator has to be upgraded.
+SRE logging requirements to date are quite stable, and the logging team can ensure they are met by new versions.
+The problem only arises if SRE want to upgrade logging for some new feature or fix, but customer refuses.
```

Comment:
True, but this is limited to configuration in the customers HLF, which are limited to forwarding audit logs.
There's a clear distinction: 
- The customer can install any version of CLO they like, and use any features they like in their CLF configurations (all logs other than API-audit logs). 
- For API audit logs, the customer must create a HLF based on the version of CLO used by SD.
So SD doesn't have to match the customer version, but does need to advertize the version they use, since that's the version for HLF resources written by the customer.  Potentially confusing but not fatal.

Reply:
The text is badly worded: "The same CLO operator will manage the customer and SRE log forwarding.", what I meant to say is "the CLO on the managed server will control log forwarding for SD and for customer API-audit logs (only)" The managed cluster is taking on some user work but only for API-audit logs, and that because we have no choice.

Reply:
fixed the text